buildifier/def.bzl

```diff
@@ -0,0 +1,62 @@
+load("@bazel_skylib//:lib.bzl", "shell")
+
+def _buildifier_impl(ctx):
+    args = [
+        "-mode",
+        ctx.attr.mode,
+    ]
+    if ctx.attr.verbose:
+        args.extend(["-v"])
+
+    exclude_path_patterns_str = ""
+    if len(ctx.attr.exclude_path_patterns):
+        exclude_path_patterns = ["-not -path %s" % shell.quote(pattern) for pattern in ctx.attr.exclude_path_patterns]
+        exclude_path_patterns_str = " ".join(exclude_path_patterns) + " -and"
+
+    out_file = ctx.actions.declare_file(ctx.label.name + ".bash")
+    substitutions = {
+        "@@ARGS@@": shell.array_literal(args),
+        "@@BUILDIFIER_SHORT_PATH@@": shell.quote(ctx.executable._buildifier.short_path),
+        "@@EXCLUDE_PATH_PATTERNS@@": exclude_path_patterns_str,
+    }
+    ctx.actions.expand_template(
+        template = ctx.file._runner,
+        output = out_file,
+        substitutions = substitutions,
+        is_executable = True,
+    )
+    runfiles = ctx.runfiles(files = [ctx.executable._buildifier])
+    return [DefaultInfo(
+        files = depset([out_file]),
+        runfiles = runfiles,
+        executable = out_file,
+    )]
+
+buildifier = rule(
+    implementation = _buildifier_impl,
+    attrs = {
+        "verbose": attr.bool(
+            doc = "Print verbose information on standard error",
+        ),
+        "mode": attr.string(
+            default = "fix",
+            doc = "Formatting mode",
+            values = ["check", "diff", "fix"],
+        ),
+        "exclude_path_patterns": attr.string_list(
```

Comment:
I'm not happy with this attribute name. Any suggestions?

Reply:
exclude_paths?
exclude_patterns?

Attributes can specify documentation (see 'doc' in https://docs.bazel.build/versions/master/skylark/lib/attr.html#string_list). Maybe mention here how it should look like.